<?xml version="1.0"?>
<launch>
    <!-- 线束 -->
    <param name="scan_line" value="16" />

    <!-- 雷达话题 -->
    <param name="/points_topic" type="string" value="/lslidar_point_cloud" />
    <!-- <param name="/points_topic" type="string" value="/kitti/velo/pointcloud" /> -->

    <!-- 是否使用gnss(室内外场景, 定位切换下使用) -->
    <param name="is_use_gnss" type="bool" value="false" />

    <!-- 是否发布tf数据 -->
    <param name="is_publish_tf" type="bool" value="true" />

    <!-- tracked frame -->
    <param name="odom_frame_id" type="string" value="map" />
    <param name="odom_child_frame_id" type="string" value="base_link" />

    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <!-- 雷达频率 -->
    <param name="scan_period" value="0.1" /> 
    <!-- 垂直分辨率 -->
    <param name="vertical_angle" type="double" value="2.0" />
    <!-- 点云距离 -->
    <param name="max_dis" type="double" value="90.0" />
    <param name="min_dis" type="double" value="1.0" />
    <!-- 特征的降采样参数：leaf -->
    <param name="map_resolution" type="double" value="0.4" />

    <!---只需要这两个节点实现定位，需要最后点云的可视化才开启第三个节点 --> 
    <node pkg="floam" type="floam_laser_processing_node" name="floam_laser_processing_node" output="screen"/>
    <node pkg="floam" type="floam_odom_estimation_node" name="floam_odom_estimation_node" output="screen"/>
    <!-- <node pkg="floam" type="floam_laser_mapping_node" name="floam_laser_mapping_node" output="screen"/> -->

    <!-- world坐标系和map坐标系 -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" /> -->

    <!-- display -->
    <arg name="rviz" default="true" /> 
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/floam.rviz" />
    </group>            
      
    <!-- 应该是调用了啥子画轨迹的东西 -->  
  	<!-- <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="velodyne" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node> -->

</launch>
